In [6]:
import pandas as pd
from scipy import stats
import numpy as np

# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing


In [7]:
data=pd.read_csv("C:\DSA\Sales_add.csv")

In [8]:
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [10]:
data[['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)']].describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [11]:
# Hypothesis
# Reject H0: There is an increase in sales after stepping into digital marketing
# Reject Ha: There is no increase in sales after stepping into digital marketing 

In [12]:
Sales_Before_Digital=data[["Sales_before_digital_add(in $)"]]
Sales_After_Digital=data[["Sales_After_digital_add(in $)"]]

In [13]:
tval,pval = stats.ttest_rel(Sales_After_Digital,Sales_Before_Digital)

In [14]:
stats.ttest_rel(Sales_After_Digital,Sales_Before_Digital)

Ttest_relResult(statistic=array([12.09070525]), pvalue=array([6.336667e-11]))

In [15]:
tval

array([12.09070525])

In [16]:
tcriteria=stats.t.ppf((1-0.05), 21)

In [17]:
stats.t.ppf((1-0.05), 21)

1.7207429028118775

In [18]:
if tval>tcriteria:
    print(' Reject H0: there is an increase in sales after stepping into digital marketing')
else:
    print('Reject Ha: There is no increase in sales after stepping into digital marketing')

 Reject H0: there is an increase in sales after stepping into digital marketing


Based on the paired(related) t tset of the sales data,before and after stepping into digital marketing, it is observed that the t value(12.09) is higher than the  critical table value of  1.7204. So there is an enough evidence to conclude that an increase in sales after stepping into digital marketing.

# The company needs to check whether there is any dependency between the Features "Region" and "Manager"


In [21]:
from scipy.stats import chi2_contingency

In [24]:
import scipy.stats

In [29]:
data1=data.rename({"Sales_before_digital_add(in $)":"SBD","Sales_After_digital_add(in $)":"SAD"},axis=1)

In [36]:
data1.head()

Month      Region      Manager     SBD     SAD
0  Month-1  Region - A  Manager - A  132921  270390
1  Month-2  Region - A  Manager - C  149559  223334
2  Month-3  Region - B  Manager - A  146278  244243
3  Month-4  Region - B  Manager - B  152167  231808
4  Month-5  Region - C  Manager - B  159525  258402

In [31]:
mycrosstable1=pd.crosstab(data1.Region,data1.Manager,values=data1.SBD,aggfunc=np.sum)

In [32]:
mycrosstable1

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A       594514       466851       420684
Region - B       570900       152167       330598
Region - C       147463       469477       130625

In [ ]:
Hypothesis
H0: There no dependancy between the features Region and Manager
Ha: There is a dependancy between the features Region and Manager

In [38]:
chiVal,pVal,df,exp=chi2_contingency(mycrosstable1)

In [39]:
chiVal,pVal,df,exp

(474356.9740349385,
 0.0,
 4,
 array([[592623.42462307, 491338.97127079, 398086.60410614],
        [421326.52881616, 349318.19201932, 283020.27916452],
        [298927.04656077, 247837.83670989, 200800.11672934]]))

In [40]:
Chi_crit=scipy.stats.chi2.ppf(1-.05, df=4)

In [41]:
Chi_crit

9.487729036781154

In [44]:
if chiVal<Chi_crit:
    print(' Accept H0: there is no dependancy between the feaures Region and Manager')
else :  
    print(' Accept Ha: there is a dependancy between the feaures Region and Manager')

 Accept Ha: there is a dependancy between the feaures Region and Manager


Based on the chi2_squired test of independance of sales data  , it is observed that chi2 value is higher than the chi_critical value of 9.48 and  rejected the  null hypothesis and  to conclude that there is a dependancy between features between Region and Manager